include/circt/Dialect/OM/OMAttributes.td

```diff
@@ -34,4 +34,22 @@ def ReferenceAttr : AttrDef<OMDialect, "Reference", [TypedAttrInterface]> {
   }];
 }
 
+def SymRefAttr : AttrDef<OMDialect, "SymRef", [TypedAttrInterface]> {
+  let summary = "An attribute that wraps a FlatSymbolRefAttr type";
+
+  let mnemonic = "symRef";
+
+  let parameters = (ins
+    "mlir::FlatSymbolRefAttr":$ref
```

Comment:
Not suggesting changes now, since we only have a use case for flat references, but maybe someday, if we need it, we can generalize to SymbolRefAttr (i.e. the one that supports nesting) in case we need to reference non-flat symbols.